src/test/scala/ai/privado/languageEngine/java/passes/read/JavaSQLDatabaseReadTest.scala

```diff
@@ -0,0 +1,293 @@
+                                             |        Connection conn = null;
+                                             |        PreparedStatement stmt = null;
+                                             |        ResultSet rs = null;
+                                             |        User user = null;
+                                             |
+                                             |        try {
+                                             |            // Connect to the database
+                                             |            conn = DriverManager.getConnection("jdbc:mysql://localhost/mydatabase", "username", "password");
+                                             |
+                                             |            // Prepare the SQL statement
+                                             |            stmt = conn.prepareStatement("SELECT firstName, lastName FROM users WHERE id = ?");
+                                             |            stmt.setInt(1, id);
+                                             |
+                                             |            // Execute the query and get the result set
+                                             |            rs = stmt.executeQuery();
+                                             |
+                                             |            // If a user with the given ID exists, create a User object and populate its fields
+                                             |            if (rs.next()) {
+                                             |                user = new User(rs.getInt("id"), rs.getString("username"), rs.getString("password"), rs.getString("email"));
+                                             |            }
+                                             |        } catch (SQLException e) {
+                                             |            e.printStackTrace();
+                                             |        } finally {
+                                             |            // Close the database resources
+                                             |            if (rs != null) rs.close();
+                                             |            if (stmt != null) stmt.close();
+                                             |            if (conn != null) conn.close();
+                                             |        }
+                                             |
+                                             |        return user;
+                                             |    }
+                                             |}
+                                             |""".stripMargin
+
+  "DatabaseReadPass" should {
+    "Tag the Select SQL Query as param" in {
+      cpg.tag.name(InternalTag.VARIABLE_REGEX_LITERAL.toString).size shouldNot (equal(0))
+    }
+
+    "Tag the SQL query with source node id" in {
+      cpg.tag.name("id").value("Data.Sensitive.FirstName").size shouldNot (equal(0))
```

Comment:
This test case is too generic as there are identifiers also which match the `FirstName` rule, you should check the tag on the node which you expect to be tagged